pkg/git.go

```diff
@@ -65,11 +87,5 @@ func (p *GitPackage) Install(ctx context.Context, dir, version string) (lockVers
 	}
 
 	commitHash := strings.TrimSpace(b.String())
-
-	err = os.RemoveAll(path.Join(dir, ".git"))
```

Comment:
I don't, but I wonder if people expect to check in the vendor directory, then this would be kind of strange. I think in the long term the .git directories will be kept in the global cache and removed in the "local" vendor directory.

This kind of is a segway to the "global cache" design I have in mind:

* Location: configurable location but by default some dot-directory in the users home directory (for example `~/.jsonnet.cache` as proposed in another comment)
* Within the global cache location, there is an index of all existing cached packages by source, version as well as resolved version (all of these variants need to be cached, instead of a single one as in this PR (but fine for the scope of this PR in my opinion). For a start one cache directory is limited/locked to a single operation at a time, this is likely enough even in the long term but can be improved if needed.
* On install/update the requested version is installed/retrieved from cache from the global cache and copied (with stripped .git directory for git source packages).

Those are my thoughts so far. What do you think @tomwilkie @metalmatze?

Reply:
Correct, what I was proposing would rather be a follow up, than a change to this PR, perhaps I should have opened a new issue to begin with. I meant that I think this change is good in the scope of this PR, but felt that since we're already having the larger discussion to share my thoughts on the global/cross-project cache.